src/utils/isPromise.ts

```diff
@@ -5,6 +5,6 @@ import type { UploadResponseFormat } from '../types/types';
  * @param  object - object to check
  * @returns
  */
-export default function isPromise(object: Promise<UploadResponseFormat>): object is Promise<UploadResponseFormat> {
+export default function isPromise<AdditionalUploadResponse>(object: Promise<UploadResponseFormat<AdditionalUploadResponse>>): object is Promise<UploadResponseFormat<AdditionalUploadResponse>> {
```

Comment:
`isPromise` is a util, so it should not depend on  `UploadResponseFormat` and `AdditionalUploadResponse`.